### PR TITLE
LEGLINK-788: Update Patient Bundle Upload

### PR DESCRIPTION
**Pull request by nvmLantana**
## Summary

This PR overhauls large-patient import behavior in `Automation.UI` and migrates imported bundle persistence to Azure Blob Storage (ABS), with deduplication and safer UX flows.

Compared to `dev/main`, this branch now:

- Stores imported patient bundle payloads in ABS (not inline Mongo JSON payloads).
- Migrates legacy inline bundle payloads to ABS in a non-blocking background process.
- Adds dedupe/remap logic for imported bundle docs and references.
- Reworks upload conflict handling to support explicit user decisions when patient data already exists on the FHIR server.
- Improves resilience and diagnostics around FHIR paging / `$everything` retrieval issues.
- Adds stronger upload UX feedback for long-running large file operations.

---

### 1) Imported bundle storage moved to ABS

- Added `IImportedBundleContentStore` abstraction and `AzureBlobImportedBundleContentStore`.
- Imported bundle content now uses `InternalBlobStorage` and is written under:
  - `BlobContainerName`
  - `BlobRoot` (`automation/imported-bundles` default in app settings).
- Dedup-aware behavior:
  - content hash (`ContentHash`) remains dedupe key.
  - blob writes skip re-upload when blob already exists.

### 2) Background migration + dedupe hardening

- Added `ImportedBundleBlobMigrationService` (hosted background migration).
- Startup-safe/non-blocking migration behavior:
  - runs after app startup.
  - batched/throttled processing.
- Migrates legacy `BundleJson` payloads from Mongo docs to ABS.
- Computes missing hashes for legacy docs.
- Deduplicates duplicate imported-bundle docs by hash.
- Remaps affected references in:
  - scenario bundle refs / scenario JSON bundle IDs
  - run input bundle IDs / run config JSON bundle IDs

### 3) Upload flow simplified for large-patient reliability

- Removed expensive/fragile full bundle deep-comparison requirement for conflict decisions.
- Upload now checks **patient existence on FHIR server** (lightweight `Patient/{id}` check).
- If patient exists, modal enforces a choice:
  1. **Purge + Replace** server data with uploaded bundle, or
  2. **Use FHIR Server Data** (discard upload, import by existing patient ID workflow), or
  3. **Cancel** (discard upload).
- For large uploads (threshold-based), UI forces the safer replace path (server-data pull option disabled), with explicit guidance.

### 4) Purge+replace operation added

- Added endpoint to purge matching server resources (`_expunge=true`) and replay uploaded bundle back to FHIR server.
- Added endpoint to discard uploaded bundle when user cancels/chooses server-data path.

### 5) UX improvements in Scenario editor

- Added upload status/progress text during file processing and classification.
- Added conflict resolution modal for existing-server-patient workflow.
- Destructive action confirm changed to **press-and-hold** pattern:
  - progress bar fill while holding
  - button turns green on completion
  - 1-second visual delay before modal closes and operation starts

### 6) FHIR loader improvements

- Added retry logic for `$everything` paging requests and clearer failure context.
- Added `PatientExistsAsync` helper for lightweight existence checks.
- Added async targeted expunge helper used by purge+replace workflow.

---

### Existing keys used (Created Automation.UI override labels for the container and root):

- `InternalBlobStorage:ConnectionString`
- `InternalBlobStorage:BlobContainerName`
- `InternalBlobStorage:BlobRoot`

### Config catalog update

- `app-config.yaml` updated to include tracking for `InternalBlobStorage:BlobRoot` (optional).

### Labeling recommendation

Use service-scoped app-config labels (e.g., `Link Automation UI`) for `InternalBlobStorage` values to isolate Automation UI bundle storage paths/containers from other services.

---

## Files with major logic changes

- `DotNet/Automation.UI/Controllers/ScenariosController.cs`
- `DotNet/Automation.UI/Views/Shared/_ScenarioEditorModal.cshtml`
- `DotNet/Automation.UI/Services/Persistence/ImportedBundleBlobMigrationService.cs`
- `DotNet/Automation.UI/Services/Persistence/AzureBlobImportedBundleContentStore.cs`
- `DotNet/Automation/FhirDataLoader.cs`
- `DotNet/Automation.UI/Services/Persistence/MongoScenarioStore.cs`

---

### 🧪 Testing Performed

Please describe the testing that was performed on the changes included in this PR.

### 🧑‍🔬 Unit Testing

- [ ] I have written or updated unit tests to cover my changes
- Coverage: 10.3%

### 📓 Documentation Updated

Please update any relevant sections in the project documentation that were impacted by the changes in the PR.

## Summary by CodeRabbit

* **New Features**
  * Large patient bundle uploads now support improved progress, validation, and status reporting.
  * Added FHIR conflict handling so you can use existing server data or replace it with uploaded patient data.
  * Added options to purge and replay patient data, or discard uploaded bundles.
  * Imported bundles are now stored securely in external blob storage, supporting larger uploads and improved retention.

* **Bug Fixes**
  * Improved handling and retry behavior for FHIR paging failures.
  * Added clearer upload and parsing error messages.

### CODE REVIEW

File: DotNet/Automation.UI/Controllers/ScenariosController.cs [hunk 1]
```diff
@@ -249,17 +282,146 @@
...
+            .OrderBy(url => url!.StartsWith("Patient/", StringComparison.OrdinalIgnoreCase) ? 1 : 0)
+            .Cast<string>()
+            .ToList();
+
+        var cfg = automationConfig.Value;
+        var loader = new FhirDataLoader(cfg.FhirServerBase, cfg.FhirServerOAuth, cfg.FhirServerBasicAuth);
+
+        logger.LogInformation(
+            "Replacing FHIR-server data for patient '{PatientId}' using uploaded bundle '{BundleId}'. Deleting {DeleteCount} resource path(s) first.",
+            request.PatientId,
```
[the remark this answers is not in the document]
Comment (nvmLantana) [reply DotNet/Automation.UI/Controllers/ScenariosController.cs]: This should be resolved, now.

**Review by johnbritton — APPROVED** [review]
